src/IntegratorLock/IntegratorLockCleaner.php

```diff
@@ -41,25 +46,31 @@ public function deleteLock(): void
     {
         $lockFilePath = $this->config->getIntegratorLockFilePath();
 
-        unlink($lockFilePath);
-
-        if (!$this->isLockFileChangeTrackedByGit($lockFilePath)) {
+        if ($this->isLockFileIgnoredByGit($lockFilePath) && !file_exists($lockFilePath)) {
             return;
         }
+        if (file_exists($lockFilePath)) {
+            unlink($lockFilePath);
+        }
+        $gitignorePath = $this->config->getProjectRootDirectory() . static::GITIGNORE_FILE;
+
+        if (!file_exists($gitignorePath) || strpos((string)file_get_contents($gitignorePath), $this->config::INTEGRATOR_LOCK) === false) {
+            file_put_contents($gitignorePath, $this->config::INTEGRATOR_LOCK . PHP_EOL, FILE_APPEND);
+        }
 
-        $this->processExecutor->execute(['git', 'add', $lockFilePath]);
-        $this->processExecutor->execute(['git', 'commit', '-m', 'Removed `integrator.lock` file.']);
+        $this->processExecutor->execute(['git', 'add', $lockFilePath, $gitignorePath]);
+        $this->processExecutor->execute(['git', 'commit', '-m', sprintf('Removed `%s` file.', $this->config::INTEGRATOR_LOCK), '-n']);
```

Comment:
Shouldn't it be `$this->config->getIntegratorLockFilePath()` instead of using static variable. As I can see in that method `$this->getProjectRootDirectory()` is used, maybe we should use it?

Reply:
everything is correct here